sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusMixClientIntegrationTest.java

```diff
@@ -0,0 +1,335 @@
+
+    ServiceBusMixClientIntegrationTest() {
+        super(new ClientLogger(ServiceBusSenderAsyncClientIntegrationTest.class));
+    }
+
+    @Override
+    protected void beforeTest() {
+        sessionId = UUID.randomUUID().toString();
+    }
+
+    @Override
+    protected void afterTest() {
+        dispose(sender);
+
+        final int numberOfMessages = messagesPending.get();
+        if (numberOfMessages < 1) {
+            dispose(receiver);
+            return;
+        }
+
+        try {
+            if (receiver == null) {
+                return;
+            }
+            receiver.receiveMessages()
+                .take(numberOfMessages)
+                .map(message -> {
+                    logger.info("Message received: {}", message.getSequenceNumber());
+                    return message;
+                })
+                .timeout(Duration.ofSeconds(5), Mono.empty())
+                .blockLast();
+        } catch (Exception e) {
+            logger.warning("Error occurred when draining queue.", e);
+        } finally {
+            dispose(receiver);
+        }
+    }
+
+    /**
+     * Use case: Test cross entity transaction using processor client and sender.
+     * 1. Read messages from entity A.
+     * 2. complete the messages from entity A and write to entity B.
+     * 2. commit the transaction.
+     */
+    @MethodSource("com.azure.messaging.servicebus.IntegrationTestBase#messagingEntityWithSessions")
+    @ParameterizedTest
+    void crossEntityTransactionWithProcessorTest(MessagingEntityType entityType, boolean isSessionEnabled) throws InterruptedException {
+
+        // Arrange
+        final boolean useCredentials = false;
+        final int receiveQueueAIndex = TestUtils.USE_CASE_TXN_1;
+        final int sendQueueBIndex = TestUtils.USE_CASE_TXN_2;
+        final String queueA = isSessionEnabled ? getSessionQueueName(receiveQueueAIndex) : getQueueName(receiveQueueAIndex);
+        final String queueB = isSessionEnabled ? getSessionQueueName(sendQueueBIndex) : getQueueName(sendQueueBIndex);
+        final String topicA = getTopicName(receiveQueueAIndex);
+        final String topicB = getTopicName(sendQueueBIndex);
+        final AtomicBoolean transactionComplete = new AtomicBoolean();
+
+        final CountDownLatch countdownLatch = new CountDownLatch(1);
+        final AtomicInteger receivedMessages = new AtomicInteger();
+
+        final String messageId = UUID.randomUUID().toString();
+        ServiceBusMessage message = TestUtils.getServiceBusMessage(CONTENTS_BYTES, messageId);
+        message.setSessionId(sessionId);
+        final List<ServiceBusMessage> messages = Arrays.asList(message);
+
+        ServiceBusClientBuilder builder = getBuilder(useCredentials).enableCrossEntityTransactions();
+
+        final ServiceBusSenderAsyncClient senderAsyncA;
+        final ServiceBusSenderClient senderSyncB;
+
+        // Initialize sender
+        switch (entityType) {
+            case QUEUE:
```

Comment:
I've realised that these tests are unwieldy. With this extra logic based on parameters. We should split these up into queue tests and subscription tests.